diagnose clusterrole: remove pods sections:

tomer-shefler opened the pull request:
all pods operations are being done in submariner-operator namespace
and thus these permessions are no longer required on the cluster role
scope

Signed-off-by: Tomer Shefler <sheflertomer@gmail.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

tomer-shefler commented:
solves https://github.com/submariner-io/subctl/issues/48
Is already possible due to https://github.com/submariner-io/subctl/pull/126

mkolesnik commented on pkg/embeddedyamls/yamls.go:
```diff
@@ -2837,14 +2837,6 @@ rules:
     verbs:
       - get
       - list
-  - apiGroups:  # pods are created to run firewall diagnostics
```
This file is auto generated, you should be editing the source file `config/rbac/submariner-diagnose/cluster_role.yaml`

skitt commented:
Thanks for submitting this! `pkg/embeddedyamls/yamls.go` is generated, please make this change in `config/rbac/submariner-diagnose/cluster_role.yaml` too (and then run `make pkg/embeddedyamls/yamls.go`).

Please fix your commit message too:

> Commit e3ba4af3d1:
> 1: T3 Title has trailing punctuation (:): "diagnose clusterrole: remove pods sections:"
> 2: B4 Second line is not empty: "all pods operations are being done in submariner-operator namespace"

skitt commented:
This would break `subctl diagnose --namespace …` for any namespace other than the operator namespace (perhaps we should remove that option).

mkolesnik commented:
> This would break `subctl diagnose --namespace …` for any namespace other than the operator namespace (perhaps we should remove that option).

We can leave the flag but document it, e.g. that when running diagnose the account used needs to have access to creating/deleting pods on the namespace if it's not the default one.

However, I think removing it is simplest as it doesn't add much value.
I think it was added to support custom operator namespace, but a more robust solution is to auto detect the operator namespace and use it for the diagnosis, without bothering the user.

tomer-shefler commented:
> > This would break `subctl diagnose --namespace …` for any namespace other than the operator namespace (perhaps we should remove that option).
> 
> We can leave the flag but document it, e.g. that when running diagnose the account used needs to have access to creating/deleting pods on the namespace if it's not the default one.
> 
> However, I think removing it is simplest as it doesn't add much value. I think it was added to support custom operator namespace, but a more robust solution is to auto detect the operator namespace and use it for the diagnosis, without bothering the user.

So, the desicion, as I understantd it, is that this namespace option shloud be remained. However a user who wants to use a different namespace should take care of all of the RBACs required to do so?

mkolesnik reviewed (approved):
> So, the desicion, as I understantd it, is that this namespace option shloud be remained. However a user who wants to use a different namespace should take care of all of the RBACs required to do so?

For me its fine as long as you add the details to the website documentation.
I'm also open for entirely removing the namespace option.

@skitt - WDYT?

mkolesnik commented:
@tomer-shefler make sure to generate the embedded yamls correctly, currently its failing the CI

tomer-shefler committed:
Remove pods sections in diagnose clusterRole

All pods operations are being done in 'submariner-operator' namespace
Thus, cluster role permissions are no longer required diagnose

Signed-off-by: Tomer Shefler <sheflertomer@gmail.com>
